### PR TITLE
Swift 2.3 and Swift 3.0 compatibility

### LINKED ISSUE #57: Swift 3 support
Any plans for swift 3 support in the near future?

### PR DESCRIPTION
**Pull request by yas375**
I was mainly interested in Swift 3.0 compatibility as I'm trying to migrate our app and its dependencies. 
Bootstrap project uses Swift 2.3

Here is what I did:

1\.  Run through migration of Bootstrap project to Swift 2.3 in Xcode 8.
2\. Have created a sample app using CocoaPods `1.1.0.rc.2` and most recent version of Quick, Nimble, FBSnapshotTestCase which support for Swift 3.0

```ruby
pod 'FBSnapshotTestCase', git: 'https://github.com/facebook/ios-snapshot-test-case.git', commit: 'da62921'
pod 'Nimble', git: 'https://github.com/Quick/Nimble.git', commit: 'db706fc'
pod 'Quick', git: 'https://github.com/Quick/Quick.git', commit: '1503fb0'
```

3\. Fork and clone Nimble-Snapshots locally
4\. In my sample app I've added Nimble-Snapshots from a local path 
```ruby
pod 'Nimble-Snapshots', path: '../Nimble-Snapshots'
```
5\. Run through migration to Swift 3 in my sample app.
6\. Having both Bootstrap and my sample app project I was jumping from one to another adding `if swift(>=3.0)` checks to make it compile with both versions.

Here you can find the sample app using Swift 3.0: 
https://github.com/yas375/ExampleApp/tree/nimble-snapshots/58-swift-3.0

Let me know if there is something else I can help with in order to get this in!

Fixes #57.

### CODE REVIEW

File: HaveValidSnapshot.swift [hunk 1]
```diff
@@ -63,15 +71,19 @@ extension UIView : Snapshotable {
 // Note that these must be lower case.
 var testFolderSuffixes = ["tests", "specs"]
 
-public func setNimbleTestFolder(testFolder: String) {
-    testFolderSuffixes = [testFolder.lowercaseString]
+public func setNimbleTestFolder(_ testFolder: String) {
+    #if swift(>=3.0)
```
Comment (orta) [inline HaveValidSnapshot.swift]: this is nice 👍

**Comment by ashfurrow** [comment]
Wow, fantastic work! I'll merge once CI is green, you reckon a release after that makes sense?

**Comment by yas375** [comment]
> Wonder if we should make a build matrix of XC7 + XC8 in travis

Good idea 👍 

I've just tried to do this, but got this error on Travis runnig Xcode8:

> “Use Legacy Swift Language Version” (SWIFT_VERSION) is required to be configured correctly for targets which use Swift. Use the [Edit > Convert > To Current Swift Syntax…] menu to choose a Swift version or use the Build Settings editor to configure the build setting directly.

It passed with Xcode 7.3 there.

I've just run `set -o pipefail && xcodebuild -workspace 'Bootstrap/Bootstrap.xcworkspace' -sdk 'iphonesimulator' -scheme 'Bootstrap' -destination "name=iPhone 6,OS=9.3" clean build test | xcpretty --color --simple` locally where I have Xcode 8 and it passes too...

And the setting it references too is actually set to 2.3

```
➜  Nimble-Snapshots git:(yas-swift3)  ack SWIFT_VERSION
Bootstrap/Bootstrap.xcodeproj/project.pbxproj
456:                            SWIFT_VERSION = 2.3;
472:                            SWIFT_VERSION = 2.3;
497:                            SWIFT_VERSION = 2.3;
518:                            SWIFT_VERSION = 2.3;

Bootstrap/Pods/Pods.xcodeproj/project.pbxproj
1226:                           SWIFT_VERSION = 2.3;
1256:                           SWIFT_VERSION = 2.3;
1286:                           SWIFT_VERSION = 2.3;
1346:                           SWIFT_VERSION = 2.3;
1377:                           SWIFT_VERSION = 2.3;
1446:                           SWIFT_VERSION = 2.3;
1481:                           SWIFT_VERSION = 2.3;
1586:                           SWIFT_VERSION = 2.3;
1616:                           SWIFT_VERSION = 2.3;
1646:                           SWIFT_VERSION = 2.3;
➜  Nimble-Snapshots git:(yas-swift3)
```

Do you guys have any ideas? Maybe you can restart the job on Travis just in case?

**Comment by ashfurrow** [comment]
I'm not sure, I've restarted the job. Looks like it could be the pods that are not updated for Xcode 8? I can help take a look tomorrow.

**Comment by yas375** [comment]
You are right. This is probably because a new version of Nimble was just released. I've tried locally to remove `Pods` and `Podfile.lock` and rerun `pod install` for the bootstrap project. Now it fails to compile in Xcode 8 for me as well.

If I understand things correctly we will need to switch to `1.1.0.rc.2` version of cocoapods because of https://github.com/CocoaPods/CocoaPods/pull/5540 And will need to add `.swift-version` with `2.3`.

Will try it right now :)

**Comment by yas375** [comment]
@orta @ashfurrow all green now! RC version of CocoaPods is in use 💚

**Comment by orta** [comment]
10/10 - this is perfect. Great job. 

@ashfurrow can you make the release?

**Comment by ashfurrow** [comment]
Ace, I'm on it.

**Comment by ashfurrow** [comment]
I've pushed a tag that you can reference manually, but I'm having a tough time pushing the pod to trunk. The pod validation fails with compiler errors from Nimble, I'm guessing because Nimble's latest version targets Swift 3 only and I'm trying to push as 2.3. However, I can't use Swift 3.0 because Quick hasn't yet released a Swift 3 version that I can reference from a podspec. 

A solution would be to limit the version of Nimble we use, but then that wouldn't work for people who _do_ want to use newer versions of Nimble.

I'm at a bit of an impasse, @orta do you have suggestions?

**Comment by yas375** [comment]
My 2 cents: as for me, I don't necessary need a pod version released for now. I'm fine with referencing it by a commit:

```ruby
pod 'Nimble-Snapshots', git: 'https://github.com/ashfurrow/Nimble-Snapshots.git', commit: '2bcf51d'
```

Maybe wait when all dependencies are released with Swift 3.0 support and release after?

**Comment by orta** [comment]
_ish_ - considering we have open deps for them both, I'd go into `which pod` find the trunk reference [for validation and remove it](https://github.com/CocoaPods/cocoapods-trunk/blob/master/lib/pod/command/trunk/push.rb#L61).

**Comment by ashfurrow** [comment]
Cool! Pushed to trunk as 4.2.0. @orta thanks for the tip, I _totally_ won't mis-use this at all  :wink:

**Pull request merged**